### PR TITLE
Resolved arrow keys clearing AI search box

### LINKED ISSUE #16471: When pressing the up or down arrow keys while entering input in Supabase AI, the input element gets cleared
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When pressing the up or down arrow keys while entering input in Supabase AI, the input element gets cleared.
Especially, when word prediction appears, I use the cursor keys, which causes the text being entered to get cleared.

https://github.com/supabase/supabase/assets/29660904/74c8dc9b-c7f2-4d98-bef0-e9841b645817

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Click on 'command K'
2. Click on 'ask supabase ai'
3. entering some chars in input form 
4. press down the up or down arrow keys

## Expected behavior

the input element is not cleared.

## System information

- OS: macOS
- Browser (if applies) chrome

### PR DESCRIPTION
**Pull request by faizanraso**
## What kind of change does this PR introduce?

This PR resolves https://github.com/supabase/supabase/issues/16471

## What is the current behavior?

Currently, pressing the up or down arrow clear will clear the input field of the AI chatbot, when no chat history exists.

https://github.com/supabase/supabase/assets/59986120/f489da9c-6525-4ecf-a91c-f4041b0109e2

## What is the new behavior?

This PR fixes this issue and does not clear the input field if there is no chat history, as shown below:

https://github.com/supabase/supabase/assets/59986120/acc535eb-3bd1-467d-89db-8f5580dbe437

### CODE REVIEW

File: packages/ui/src/components/Command/Command.utils.tsx [hunk 1]
```diff
@@ -358,14 +358,14 @@ export function useHistoryKeys({ enable, messages, setPrompt }: UseHistoryKeysOp
         case 'ArrowUp':
           setMessageSelectionIndex((index) => {
             const newIndex = Math.max(index - 1, 0)
-            setPrompt(messages[newIndex] ?? '')
+            messages[newIndex] ? setPrompt(messages[newIndex]) : null
```
Comment (gregnr) [inline packages/ui/src/components/Command/Command.utils.tsx:361]: Since there's no assignment, lets use an `if` statement for better clarity.

**Review by gregnr — COMMENTED** [review]
Thanks @faizanraso! Just a small change then we'll merge this.

File: packages/ui/src/components/Command/Command.utils.tsx (hunk 1 above)
Comment (faizanraso) [reply packages/ui/src/components/Command/Command.utils.tsx]: Hi @gregnr I've pushed the changes. Thanks for the feedback!

File: packages/ui/src/components/Command/Command.utils.tsx [hunk 2]
```diff
@@ -358,14 +358,14 @@ export function useHistoryKeys({ enable, messages, setPrompt }: UseHistoryKeysOp
         case 'ArrowUp':
           setMessageSelectionIndex((index) => {
             const newIndex = Math.max(index - 1, 0)
-            setPrompt(messages[newIndex] ?? '')
+            if (messages[newIndex]) setPrompt(messages[newIndex])
```
Comment (gregnr) [inline packages/ui/src/components/Command/Command.utils.tsx:361]: For additional clarity & prevent future errors:
```suggestion
            const newMessage = messages[newIndex]
            if (newMessage) {
              setPrompt(newMessage)
            }
```

File: packages/ui/src/components/Command/Command.utils.tsx [hunk 3]
```diff
@@ -358,14 +358,14 @@ export function useHistoryKeys({ enable, messages, setPrompt }: UseHistoryKeysOp
...
             return newIndex
           })
           return
         case 'ArrowDown':
           setMessageSelectionIndex((index) => {
             const newIndex = Math.min(index + 1, messages.length)
-            setPrompt(messages[newIndex] ?? '')
+            if (messages[newIndex]) setPrompt(messages[newIndex])
```
Comment (gregnr) [inline packages/ui/src/components/Command/Command.utils.tsx:368]:
```suggestion
            const newMessage = messages[newIndex]
            if (newMessage) {
              setPrompt(newMessage)
            }
```

**Review by gregnr — APPROVED** [review]

**Pull request merged**